src/libs/common/types/DecorationRangeBehavior.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * Describes the behavior of decorations when typing/editing at their edges.
+ */
+export type DecorationRangeBehavior =
```

Comment:
Would it be worth making this a string `const enum`? It seems a bunch of changes are just changing `DecorationRangeBehavior` references to strings.

Reply:
We are not really using enums in cursorless and are iinstead using typed strings. I'm not against enums, but then I think we should use them in all places we can. Ie I prefer one solution or the other.

Reply:
I'm increasingly leaning towards enums, because then we can do renames, and we have proper JSDoc support for the members.  Shall we use these as our first enums, and then start migrating the others?  I don't think we need to migrate all strings to enums at once.  I think we probably want to keep strings on our API surface, unfortunately, though maybe we can use them there as well?

Maybe let's discuss at meet-up

Reply:
Discussed at meet-up and we want to start using enums instead of strings, for the reasons listed above, and we plan to roll back these enum-to-string changes in the PR, and start migrating our other strings to enums in follow up PRs

Reply:
Ok switched this and `RevealLineAt` to be enums.  I used integers for the former, and strings for the latter.  I think integers are ok internally, but for api surface we should use strings.  Because we pass `RevealLineAt` to the ide api, it will end up in lsp api.  Could be convinced to just use string enums everywhere 🤷‍♂️